src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php

```diff
@@ -403,13 +398,14 @@ public function close()
 
             // delete the session records that have expired
             if ('mysql' === $this->driver) {
-                $sql = "DELETE FROM $this->table WHERE $this->lifetimeCol + $this->timeCol < :time";
+                $sql = "DELETE FROM $this->table WHERE ($this->lifetimeCol < :time AND $this->lifetimeCol > :min) OR ($this->lifetimeCol + $this->timeCol < :time)";
```

Comment:
using `OR` means that the DB engine won't use the index AFAIK
that's why I suggested using two queries